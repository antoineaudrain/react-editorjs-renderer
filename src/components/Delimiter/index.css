.er-delimiter {
    width: 100%;
    line-height: 1.6em;
    text-align: center;
    padding: .4em 0;
}

.er-delimiter::before {
    display: inline-block;
    content: '***';
    font-size: 30px;
    line-height: 65px;
    height: 30px;
    letter-spacing: 0.2em;
    color: #313649;

}