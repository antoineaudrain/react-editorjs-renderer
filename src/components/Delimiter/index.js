import React from 'react'
import PropTypes from 'prop-types'
import propsConstructor from '../../tools/propsConstructor'
import defaultStyle from './index.style'

const Delimiter = ({ style = {} }) => {
  const getProps = propsConstructor(style, defaultStyle)
  const delimiterStyle = getProps('delimiter')

  return <div {...delimiterStyle} />
}

Delimiter.propTypes = {
  style: PropTypes.object
}

export default Delimiter
