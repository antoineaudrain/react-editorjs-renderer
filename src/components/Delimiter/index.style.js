const style = {
  width: '100%',
  lineHeight: '1.6em',
  textAlign: 'center',
  padding: '.4em 0',

  '&::before': {
    display: 'inline-block',
    content: '***',
    fontSize: '30px',
    lineHeight: '65px',
    height: '30px',
    letterSpacing: '0.2em',
    color: '#313649'
  }
}

export default style