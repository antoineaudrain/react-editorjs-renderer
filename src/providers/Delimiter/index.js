import React from 'react'
import PropTypes from 'prop-types'
import providerPropsConstructor from '../../utils/providerPropsConstructor'
import defaultStyle from './index.style'

const Delimiter = ({ style = {}, config = {} }) => {
  const getProps = providerPropsConstructor(defaultStyle, style, config)
  const delimiterStyle = getProps('delimiter')

  return <div {...delimiterStyle} />
}

Delimiter.propTypes = {
  style: PropTypes.object,
  config: PropTypes.object
}

export default Delimiter
