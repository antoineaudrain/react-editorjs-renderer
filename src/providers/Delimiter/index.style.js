const style = {
  delimiter: {
    width: '100%',
    linHeight: '1.6em',
    textAlign: 'center',
    padding: '.4em 0'
  },
  'delimiter::before': {
    display: 'inline-block',
    content: '***',
    fontSize: '30px',
    lineHeight: '65px',
    height: '30px',
    letterSpacing: '0.2em'
  }
}

export default style
