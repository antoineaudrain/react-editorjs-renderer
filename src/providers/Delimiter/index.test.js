import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Delimiter from './index'

configure({ adapter: new Adapter() })

describe('Header provider:', () => {
  const delimiter = shallow(<Delimiter />)
  const disabledStyleDelimiter = shallow(
    <Delimiter config={{ disableDefaultStyle: true }} />
  )

  it('should match snapshot', () => expect(delimiter).toMatchSnapshot())
  it('should match snapshot', () =>
    expect(disabledStyleDelimiter).toMatchSnapshot())
})
