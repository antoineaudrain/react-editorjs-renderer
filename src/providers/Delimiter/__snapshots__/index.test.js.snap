// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header provider: should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Header provider: should match snapshot 2`] = `ShallowWrapper {}`;
